### PR TITLE
HEEDLS-705 Change calculation for role requirements value to look at …

### PR DESCRIPTION
**Pull request by AlexJacksonDS**
…more specific record

### JIRA link
https://softwiretech.atlassian.net/browse/HEEDLS-705

### Description
Changed the role requirement score calculation so that the record is more specified as per the requirements change.

### Screenshots
N/A

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [x] Run the formatter and made sure there are no IDE errors.
- [x] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [ ] Manually tested my work with and without JavaScript. Full manual testing guidelines can be found here: https://softwiretech.atlassian.net/wiki/spaces/HEE/pages/6703648740/Testing
- [ ] Updated/added documentation in Swiki and/or Readme. Links (if any) are below:
  - [doc_1_here](link_1_here)
- [ ] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken.
- [x] Scanned over my own MR to ensure everything is as expected.

### CODE REVIEW

**Review by DanBloxham-sw — CHANGES_REQUESTED** [review]
Looks pretty good, one minor comment about a parameter name.

**Review by DanBloxham-sw — APPROVED** [review]

File: DigitalLearningSolutions.Data.Tests/Services/RecommendedLearningServiceTests.cs [hunk 1]
```diff
@@ -430,6 +437,8 @@ public async Task
             A.CallTo(
                 () => selfAssessmentDataService.GetCompetencyAssessmentQuestionRoleRequirements(
                     A<int>._,
+                    A<int>._,
+                        A<int>._,
```
Comment (SteveJacksonSoft) [inline DigitalLearningSolutions.Data.Tests/Services/RecommendedLearningServiceTests.cs:441]: This looks misaligned

**Review by SteveJacksonSoft — COMMENTED** [review]
Looks good, just noticed a formatting issue on one line.

File: DigitalLearningSolutions.Data.Tests/Services/RecommendedLearningServiceTests.cs (hunk 1 above)
Comment (SteveJacksonSoft) [reply DigitalLearningSolutions.Data.Tests/Services/RecommendedLearningServiceTests.cs:441]: I'm happy for you to merge without rereview unless you want Olivia's judgement.

**Pull request merged**